spec/requests/well_known/webfinger_spec.rb

```diff
@@ -0,0 +1,255 @@
+    let(:resource) do
+      "#{Rails.configuration.x.local_domain}@#{Rails.configuration.x.local_domain}"
+    end
+
+    before do
+      perform_request!
+    end
+
+    it 'returns http success' do
+      expect(response).to have_http_status(200)
+    end
+
+    it 'sets only a Vary Origin header' do
+      expect(response.headers['Vary']).to eq('Origin')
+    end
+
+    it 'returns application/jrd+json' do
+      expect(response.media_type).to eq 'application/jrd+json'
+    end
+
+    it 'returns links for the internal account' do
+      json = body_as_json
+      expect(json[:subject]).to eq 'acct:mastodon.internal@cb6e6126.ngrok.io'
+      expect(json[:aliases]).to eq ['https://cb6e6126.ngrok.io/actor']
+    end
+  end
+
+  context 'with no resource parameter' do
+    let(:resource) { nil }
+
+    before do
+      perform_request!
+    end
+
+    it 'returns http bad request' do
+      expect(response).to have_http_status(400)
+    end
+  end
+
+  context 'with a nonsense parameter' do
+    let(:resource) { 'df/:dfkj' }
+
+    before do
+      perform_request!
+    end
+
+    it 'returns http bad request' do
+      expect(response).to have_http_status(400)
+    end
+  end
+
+  context 'when an account has an avatar' do
+    let(:alice) { Fabricate(:account, username: 'alice', avatar: attachment_fixture('attachment.jpg')) }
+    let(:resource) { alice.to_webfinger_s }
+
+    it 'returns avatar in response' do
+      perform_request!
+
+      avatar_link = get_avatar_link(body_as_json)
+      expect(avatar_link).to_not be_nil
+      expect(avatar_link[:type]).to eq alice.avatar.content_type
+      expect(avatar_link[:href]).to eq Addressable::URI.new(host: Rails.configuration.x.local_domain, path: alice.avatar.to_s, scheme: 'https').to_s
+    end
+
+    context 'with limited federation mode' do
+      before do
+        allow(Rails.configuration.x).to receive(:limited_federation_mode).and_return(true)
+      end
+
+      it 'does not return avatar in response' do
+        perform_request!
+
+        avatar_link = get_avatar_link(body_as_json)
+        expect(avatar_link).to be_nil
+      end
+    end
+
+    context 'when enabling DISALLOW_UNAUTHENTICATED_API_ACCESS' do
+      around do |example|
+        ClimateControl.modify DISALLOW_UNAUTHENTICATED_API_ACCESS: 'true' do
+          example.run
+        end
+      end
+
+      it 'does not return avatar in response' do
+        perform_request!
+
+        avatar_link = get_avatar_link(body_as_json)
+        expect(avatar_link).to be_nil
+      end
+    end
+  end
+
+  context 'when an account does not have an avatar' do
+    let(:alice) { Fabricate(:account, username: 'alice', avatar: nil) }
+    let(:resource) { alice.to_webfinger_s }
+
+    before do
+      perform_request!
+    end
+
+    it 'does not return avatar in response' do
+      avatar_link = get_avatar_link(body_as_json)
+      expect(avatar_link).to be_nil
+    end
+  end
+
+  context 'with different headers' do
```

Comment:
This whole block moves in what was previously the `spec/requests/webfinger_request_spec` checks